### ISSUE TITLE
Work around Oracle / ojdbc bug where JSON_ARRAY() can't combine bind parameter marker with FORMAT JSON

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13156]
Since jOOQ 3.15.0 and https://github.com/jOOQ/jOOQ/issues/12080, we generate a `FORMAT JSON` clause when nesting JSON expressions in other JSON expressions, as required by Oracle.

Unfortunately, there seems to be an ojdbc bug where bind variables shortly after `FORMAT JSON` aren't recognised:

```java
try (PreparedStatement s = connection.prepareStatement(
    """
    select
      json_arrayagg(
        json_array(
          (select json_arrayagg(json_array(?) format json) from dual) format json,
          ?
        )
        format json
      )
    from dual
    """
)) {
    s.setInt(1, 1);
    s.setInt(2, 2); // This fails

    try (ResultSet rs = s.executeQuery()) {
        while (rs.next())
            System.out.println(rs.getString(1));
    }
}
```

This fails with:

```
Exception in thread "main" java.sql.SQLException: Invalid column index
    at oracle.jdbc.driver.OraclePreparedStatement.setIntInternal(OraclePreparedStatement.java:4956)
    at oracle.jdbc.driver.OraclePreparedStatement.setInt(OraclePreparedStatement.java:4947)
    at oracle.jdbc.driver.OraclePreparedStatementWrapper.setInt(OraclePreparedStatementWrapper.java:202)
    at org.jooq.testscripts.JDBC.main(JDBC.java:50)
```

I've documented a few workarounds here: https://stackoverflow.com/q/71268781/521799. Wrapping the bind value with `NVL(NULL, ?)` seems to work best. The `CAST(? AS <type>)` approach might cause trouble when we choose the wrong data type (e.g. `VARCHAR2(4000)` instead of `CLOB`)

----
A full description of the bug can be seen here:
https://stackoverflow.com/q/71268781/521799

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.5 (https://github.com/jOOQ/jOOQ/issues/13158), and 3.15.9 (https://github.com/jOOQ/jOOQ/issues/13159)

**Comment by lukaseder** [comment]
This issue seems to have been fixed in Oracle 23c / ojdbc `23.3.0.23.09`:

- https://github.com/jOOQ/jOOQ/issues/15835
